drivers/overlay/ov_network.go

```diff
@@ -296,48 +295,62 @@ func (d *driver) RevokeExternalConnectivity(nid, eid string) error {
 	return nil
 }
 
-func (n *network) incEndpointCount() {
-	n.Lock()
-	defer n.Unlock()
-	n.joinCnt++
-}
+func (n *network) joinSandbox(s *subnet, restore bool, incJoinCnt bool) error {
+	networkOnce.Do(networkOnceInit)
+
+	n.joinLock.Lock()
+	defer n.joinLock.Unlock()
 
-func (n *network) joinSandbox(restore bool) error {
-	// If there is a race between two go routines here only one will win
-	// the other will wait.
-	n.once.Do(func() {
-		// save the error status of initSandbox in n.initErr so that
-		// all the racing go routines are able to know the status.
+	if !n.sboxInit {
 		n.initErr = n.initSandbox(restore)
-	})
+		n.sboxInit = true // we cannot recover from this error
+	}
 
-	return n.initErr
-}
+	if n.initErr != nil {
+		return fmt.Errorf("network sandbox join failed: %v", n.initErr)
+	}
 
-func (n *network) joinSubnetSandbox(s *subnet, restore bool) error {
-	s.once.Do(func() {
-		s.initErr = n.initSubnetSandbox(s, restore)
-	})
-	return s.initErr
+	subnetErr := s.initErr
+	if !s.sboxInit {
```

Comment:
isn't this dead code? 
if at line 304 was false, will be set to true
if it was already true then won't enter here anyway

Reply:
This is `s.sboxInit` not `n.sboxInit` ...  So no, this is different. :)